docs/developer-guide/working-with-plugins.md

```diff
@@ -66,6 +66,28 @@ module.exports = {
 };
 ```
 
+### Custom Environments in Plugins
+
+Plugins can also provide custom environments. Custom environments can declare `globals`, turn `ecmaFeatures` on and off,
+and configure both core rules as well as rules included in the plugin. To provide custom environment, include `environment`
+property in your exported object.
+
+```js
+module.exports = {
+    environments: {
+        "my-custom-environment": {
+            globals: {
+                "global-variable": true
+            },
+            rules: {
+                "no-alert": 0, //turn off no-alert rule
+                "my-plugin/my-rule": 1 //turn on custom rule from this plugin (given that plugin's name is "my-plugin")
```

Comment:
This is an interesting decision. The rules themselves in the plugin are only referred to be their local name (`my-rule` here) when exported. I suppose we need to use the plugin name here so we can also adjust native rules...or should we?

This would be the only spot in the plugin where you'd need to specify the plugin name. Does it make sense to do so? Should it be more generic, like `*/my-rule` so you don't have to write the full plugin name and then it's inserted automatically for you?